client/components/ChatInput.vue

```diff
@@ -337,6 +337,7 @@ export default defineComponent({
 				autocompletionRef.value = undefined;
 			}
 
+			upload.unmounted();
 			upload.abort();
```

Comment:
Would it make sense to fold abort into unmounted?

As in, calling unmount should make the uploader clean up after itself no?
Seems strange to shit that to the caller after it told the uploader that it should go away

Reply:
shift* ah, bad typo... sorry

Reply:
I guess we could make `unmounted` call `abort`, but this whole thing kind of needs refactoring honestly. The way the pasting of the link works completely bypasses Vue. As the comment says, abort is a hack already.

One could want to call `abort` on pressing escape, and one could want to stop accepting new uploads but keep the current one going in the background, although right now that's undefined behaviour that crashes.

Ideally one should be able to start pasting files into different channels and not interfere, but right now it just gets pasted into whatever is active right now despite otherwise being able to start typing in one channel, move to another, type, move back and continue typing the original message. If bound the upload to the channel buffer rather than the input field, we wouldn't need abort and could register/deregister as needed.

Reply:
yeah... I see the problem.

One more thing on the todo list I guess.
Thanks.